mailet/icalendar/src/main/java/org/apache/james/transport/mailets/ICALToJsonAttribute.java

```diff
@@ -170,27 +169,38 @@ public void service(Mail mail) throws MessagingException {
     }
 
     private void setAttribute(Mail mail, Map<String, Calendar> calendars, Map<String, byte[]> rawCalendars) throws MessagingException {
-        Optional<String> sender = retrieveSender(mail);
+        Optional<MailAddress> sender = retrieveSender(mail);
         if (!sender.isPresent()) {
             LOGGER.info("Skipping {} because no sender and no from", mail.getName());
             return;
         }
+
+        MailAddress effectiveSender = sender.get();
+        MailAddress participant = AttributeUtils.getValueAndCastFromMail(mail, REPLY_TO_ATTRIBUTE, String.class)
+                .map(MaybeSender::getMailSender)
+                .flatMap(MaybeSender::asOptional)
+                .orElse(effectiveSender);
```

Comment:
IMO we should not assume in James code that the replyTo is the participant. We should let OpenPaaS take this decision and just pass it the info.